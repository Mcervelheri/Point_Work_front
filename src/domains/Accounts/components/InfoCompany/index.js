import React, { useCallback, useMemo } from 'react';

import { Card } from 'antd';

import Icon, { AntDesign } from '../../../../components/Icon';

import styles from './styles.module.scss';

const InfoCompany = ({ onEdit, accountCompany }) => {

    const handleEdit = useCallback(() => {
        onEdit('COMPANY');
    }, [onEdit]);

    const actionsIcon = useMemo(() => ([
        <Icon
            family={AntDesign}
            name="FillEdit"
            size={16}
            onClick={handleEdit}
        />,
    ]), [handleEdit]);

    if (!accountCompany) return null;

    return (
        <Card
            actions={actionsIcon}
            className={styles.container}
        >
            <p className={styles.nameCompany}>{accountCompany?.company?.tradeName}</p>
            <p className={styles.cnpjCompany}>{accountCompany?.company?.nationalRegistration}</p>
        </Card>
    );
};

export default InfoCompany;
