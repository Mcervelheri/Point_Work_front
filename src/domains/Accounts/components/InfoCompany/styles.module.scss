@import "../../../../theme/colors.scss";
@import "../../../../theme/variables.scss";

.container {
    margin-bottom: 32px;
}

.nameCompany {
    font-weight: normal;
    font-size: $large-font;
    line-height: 24px;
}

.cnpjCompany {
    font-weight: normal;
    font-size: $medium-font;
    line-height: 22px;
    color: rgba($color: $color-gray-13, $alpha: 0.45); 
    margin-top: -8px;
}
