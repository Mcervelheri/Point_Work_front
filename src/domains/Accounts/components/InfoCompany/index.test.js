import React from 'react';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import InfoCompany from './index';

const mockStore = configureMockStore();
const store = mockStore({});

const MOCK_INFO_COMPANY = {
    id: 1,
    name: 'Fazenda Vaquinha Feliz Ltda.',
    cnpj: '00.000.000/0001-00',
};

test('O Componente <InfoCompany /> deve ser renderizado na tela corretamente', () => {
    const Element = renderer.create(
        <Provider store={store}>
            <InfoCompany
                name={MOCK_INFO_COMPANY.name}
                cpf={MOCK_INFO_COMPANY.cnpj}
            />
        </Provider>,
    ).toJSON();
    expect(Element).toMatchSnapshot();
});
