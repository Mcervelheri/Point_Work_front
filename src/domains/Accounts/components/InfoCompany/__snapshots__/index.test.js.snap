// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O Componente <InfoCompany /> deve ser renderizado na tela corretamente 1`] = `null`;
